src/Algebra/Properties/CommutativeMonoid.agda

```diff
@@ -0,0 +1,181 @@
+--
+-- Some derivable properties
+------------------------------------------------------------------------
+
+open import Algebra
+
+module Algebra.Properties.CommutativeMonoid {g₁ g₂} (M : CommutativeMonoid g₁ g₂) where
+
+open CommutativeMonoid M renaming (ε to 0#; _∙_ to _+_; ∙-cong to +-cong; identity to +-identity; assoc to +-assoc; comm to +-comm)
+open import Algebra.Operations.CommutativeMonoid M
+import Algebra.FunctionProperties as Props; open Props _≈_
+import Relation.Binary.EqReasoning as EqR; open EqR setoid
+import Relation.Binary as B
+open import Function
+open import Function.Equality using (_⟨$⟩_)
+open import Data.Product
+import Data.Bool as Bool
+open import Data.Nat using (ℕ; zero; suc)
+open import Data.Fin using (Fin; punchIn; zero; suc)
+open import Data.List as List using ([]; _∷_)
+import Data.Fin.Properties as FP
+open import Data.Fin.Permutation as Perm using (Permutation; Permutation′; _⟨$⟩ˡ_; _⟨$⟩ʳ_)
+open import Data.Fin.Permutation.Components as PermC
+open import Data.Table as Table
+open import Data.Table.Relation.Equality as TE using (_≗_)
+open import Data.Unit using (tt)
+open import Data.Empty using (⊥-elim)
+import Data.Table.Properties as TP
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+open import Relation.Nullary using (yes; no)
+open import Relation.Nullary.Decidable using (⌊_⌋)
+
+module _ {n} where
+  open B.Setoid (TE.setoid setoid n) public
+    using ()
+    renaming (_≈_ to _≋_)
+
+-- When summing over a function from a finite set, we can pull out any value and move it to the front.
+
+sumₜ-punchIn : ∀ {n} t (i : Fin (suc n)) → sumₜ t ≈ lookup t i + sumₜ (rearrange (punchIn i) t)
+sumₜ-punchIn f zero = refl
+sumₜ-punchIn {zero} t (suc ())
+sumₜ-punchIn {suc n} t (suc i) =
+  let x = head t
+      y = lookup t (suc i)
+      z = sumₜ (rearrange (punchIn i) (tail t))
+  in begin
+    x + sumₜ (tail t)  ≈⟨ +-cong refl (sumₜ-punchIn (tail t) i) ⟩
+    x + (y + z)        ≈⟨ sym (+-assoc _ _ _) ⟩
+    (x + y) + z        ≈⟨ +-cong (+-comm _ _) refl ⟩
+    (y + x) + z        ≈⟨ +-assoc _ _ _ ⟩
+    y + (x + z)        ∎
+
+-- '_≈_' is a congruence over 'sumTable n'.
+
+sumₜ-cong : ∀ {n} {t t′ : Table Carrier n} → t ≋ t′ → sumₜ t ≈ sumₜ t′
+sumₜ-cong {zero} p = refl
+sumₜ-cong {suc n} p = +-cong (p _) (sumₜ-cong (p ∘ suc))
+
+-- '_≡_' is a congruence over 'sum n'.
+
+sumₜ-cong≡ : ∀ {n} {t t′ : Table Carrier n} → t ≗ t′ → sumₜ t ≡ sumₜ t′
+sumₜ-cong≡ {zero} p = P.refl
+sumₜ-cong≡ {suc n} p = P.cong₂ _+_ (p _) (sumₜ-cong≡ (p ∘ suc))
+
+-- The sum over the constantly zero function is zero.
+
+sumₜ-zero : ∀ n → sumₜ (replicate {n} 0#) ≈ 0#
+sumₜ-zero (zero) = refl
+sumₜ-zero (suc n) =
+  begin
+    0# + sumₜ (replicate {n} 0#)  ≈⟨ proj₁ +-identity _ ⟩
+    sumₜ (replicate {n} 0#)       ≈⟨ sumₜ-zero n ⟩
+    0#                            ∎
+
+-- The '∑' operator distributes over addition.
+
+∑-+-hom : ∀ n (f g : Fin n → Carrier) → ∑[ i < n ] f i + ∑[ i < n ] g i ≈ ∑[ i < n ] (f i + g i)
```

Comment:
I think the name`∑-distrib-+` might be better for this. Perhaps(?) clearer and it follows the conventions set out in `Nat.Properties` etc.